docs/joss/paper.md

```diff
@@ -27,50 +28,141 @@ affiliations:
+sparse linear algebra operations. While GPGPU utilization for high-performance linear algebra is common, 
+the high complexity of GPGPU programming makes the implementation of the complete set of sparse operations on GPGPU challenging. 
+Thus, it is worth addressing this problem by focusing on a basic but still important case — sparse Boolean algebra.
+
+The primary goal of the `SPbLA` is to provide a base for the implementation, 
+testing and profiling high-performance algorithms for solving data analysis problems, 
+such as RDF analysis [@article:cfpq_and_rdf_analysis], 
+RNA secondary structure analysis [@article:rna_prediction], 
+static program code analysis (such as points-to or alias analysis) [@article:dyck_cfl_code_analysis] 
 and evaluation of regular and CFL-reachability queries [@inproceedings:matrix_cfpq; @inbook:kronecker_cfpq_adbis]. 
 
-Also we hope, that the library is a small step to move forward the implementation of 
-the fully featured sparse linear algebra with generalization to arbitrary
-monoids and searings for multi-GPU computations.
+Thus, we can offload different language-constrained path querying related problems, 
+and other problems that can be reduced to manipulation of boolean matrices, to GPGPU in a uniform way. 
+
+Moreover, real data analysis leads to huge matrix processing that can not be efficiently 
+handled on a single GPGPU. Thus the creation of the library which supports multi-GPU and 
+out-of-VRAM computations helps to create an efficient solution for a wide range of applied problems. 
+The creation of such a solution is an open problem while ad-hoc solutions exist in specific areas. 
+We propose an SPbLA as a base for such a solution.
+
+Also, we hope, that the library is a small step to move forward the implementation of 
+the fully-featured sparse linear algebra as specified in GrpahBLAS multi-GPU computations.
+
 
 # Related tools
 
+`GraphBLAS API` [@paper:graphblas_foundations] is one of the first standards, that
+formalize the mathematical building blocks in the form of the programming interface
+for implementing algorithms in the language of the linear algebra. 
+`SuiteSparse` [@article:suite_sparse_for_graph_problems] is a reference implementations
+of the `GraphBLAS API` for CPU computation. It is mature and fully featured library
+with number of bindings for other programming languages, such `pygraphblas` [@pygraphblas] 
+for Python programming.
+
+GPGPU's utilization for data analysis and for linear algebra operations is a promising 
+way to high-performance data analysis because GPGPU is much more powerful in parallel
+data processing. However, GPGPU programming is still challenging.
+Best to our knowledge, there is no complete GraphBLAS API implementation for GPGPU
+computations, except `GraphBLAST` [@yang2019graphblast], which is currently in the
+active development. Some work is also done to move SuiteSparse forward GPGPU computations.
+OpenCL and portability?
```

Comment:
What do we want to write here? Tools problems, portablility issues, CUDA vs OpenCL?

Reply:
I guess we can just remove the last sentence and replace it with some conclusion (or with a transition sentence to the next paragraph).